zrml/swaps/fuzz/utils.rs

```diff
@@ -0,0 +1,201 @@
+#![allow(
+    // Mocks are only used for fuzzing and unit tests
+    clippy::integer_arithmetic
+)]
+
+use zeitgeist_primitives::constants::{
+    MaxAssets, MaxTotalWeight, MaxWeight, MinAssets, MinLiquidity, MinWeight, BASE,
+};
+
+use arbitrary::{Arbitrary, Result, Unstructured};
+
+use rand::{rngs::ThreadRng, seq::SliceRandom, Rng};
+
+use zeitgeist_primitives::types::{Asset, ScalarPosition, SerdeWrapper};
+
+use zeitgeist_primitives::{
+    traits::Swaps as SwapsTrait,
+    types::{PoolId, ScoringRule},
+};
+use zrml_swaps::mock::Swaps;
+
+pub fn construct_asset(seed: (u128, u16)) -> Asset<u128> {
+    let (seed0, seed1) = seed;
+    let module = seed0 % 4;
+    match module {
+        0 => Asset::CategoricalOutcome(seed0, seed1),
+        1 => {
+            let scalar_position =
+                if seed1 % 2 == 0 { ScalarPosition::Long } else { ScalarPosition::Short };
+            Asset::ScalarOutcome(seed0, scalar_position)
+        }
+        2 => Asset::CombinatorialOutcome,
+        3 => Asset::PoolShare(SerdeWrapper(seed0)),
+        _ => Asset::Ztg,
+    }
+}
+
+#[derive(Debug)]
+pub struct ValidPoolData {
+    pub origin: u128,
+    pub assets: Vec<(u128, u16)>,
+    pub base_asset: (u128, u16),
+    pub market_id: u128,
+    pub swap_fee: u128,
+    pub amount: u128,
+    pub weights: Vec<u128>,
+}
+
+impl ValidPoolData {
+    pub fn _create_pool(self) -> PoolId {
```

Comment:
In that case I get `associated function is never used: 'create_pool'`. Clippy doesn't  pass then. Do you know a solution to that? I mean the `create_pool` function is indeed in use. But I don't really know, why rust doesn't recognize this. Is it because there is a file separation or has it something to do with the executable environment in which extrinsics are run?

Reply:
Ok, `cargo check` does detect that `create_pool` is not used. `cargo clippy` detects it too with `error: associated function is never used: 'create_pool'`. Instead of `_create_pool` I now used `#[allow(dead_code)]`. Thanks for the suggestion.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/542/commits/3ea49870bcdf5895e19d3dd6e376e1ac095c917e